src/entry.ts

```diff
@@ -182,6 +200,45 @@ class Entry {
     return entry;
   }
 
+  /**
+   * Formats raw incoming request objects into a GCP structured HTTPRequest.
+   * Formats trace & span if users provided X-Cloud-Trace-Context in header.
+   * See more: https://cloud.google.com/trace/docs/setup#force-trace
+   *    "X-Cloud-Trace-Context: TRACE_ID/SPAN_ID;o=TRACE_TRUE"
+   * for example:
+   *    "X-Cloud-Trace-Context: 105445aa7843bc8bf206b120001000/1;o=1"
+   * Note: logs from middleware are already formatted.
+   *
+   * @private
+   */
+  private formatHttpRequest() {
+    const rawReq = this.metadata.httpRequest;
+    if (rawReq) {
+      // Handle raw http requests.
+      if (
+        'statusCode' in rawReq ||
+        'headers' in rawReq ||
+        'method' in rawReq ||
+        'url' in rawReq
+      )
+        this.metadata.httpRequest = makeHttpRequestData(rawReq);
+      // Infer trace & span if not user specified already.
+      if ('headers' in rawReq && rawReq.headers['x-cloud-trace-context']) {
+        const regex = /([a-f\d]+)?(\/?([a-f\d]+))?(;?o=(\d))?/;
```

Comment:
I suggest you refactor this to prepare for recognition of trace/span from traceparent (the recognition itself might happen in a different PR).

Reply:
ah ok, ill do this in a following pr with a TODO note on recognizing `trace-parent`